pkg/backup/backup/backup_cleaner.go

```diff
@@ -52,7 +53,81 @@ func NewBackupCleaner(deps *controller.Dependencies, statusUpdater controller.Ba
 }
 
 func (bc *backupCleaner) Clean(backup *v1alpha1.Backup) error {
-	if backup.DeletionTimestamp == nil || !v1alpha1.IsCleanCandidate(backup) || v1alpha1.NeedNotClean(backup) {
+	if backup.DeletionTimestamp == nil {
+		// The backup object has not been deleted，do nothing
+		return nil
+	}
+
+	klog.Infof("start to clean backup %s/%s", backup.GetNamespace(), backup.GetName())
+	if err := bc.CleanLogBackup(backup); err != nil {
+		return err
+	}
+	return bc.CleanData(backup)
+}
+
+func (bc *backupCleaner) CleanLogBackup(backup *v1alpha1.Backup) error {
+	if backup.Spec.Mode != v1alpha1.BackupModeLog {
+		return nil
+	}
+	if backup.Spec.BR == nil {
+		return fmt.Errorf("backup %s/%s spec.BR shouldn't be nil", backup.GetNamespace(), backup.GetName())
+	}
+	if !v1alpha1.IsLogBackupAlreadyStart(backup) {
+		return nil
+	}
+	if v1alpha1.IsLogBackupAlreadyStop(backup) {
+		return nil
+	}
+
+	ns := backup.GetNamespace()
+	name := backup.GetName()
+	stopLogJobName := backup.GetCleanLogBackupJobName()
+
+	var err error
+	_, err = bc.deps.JobLister.Jobs(ns).Get(stopLogJobName)
+	if err == nil {
+		// already have a clean job running，return directly
+		return nil
+	} else if !errors.IsNotFound(err) {
+		bc.statusUpdater.Update(backup, &v1alpha1.BackupCondition{
+			Type:    v1alpha1.BackupRetryTheFailed,
+			Status:  corev1.ConditionTrue,
+			Reason:  "GetBackupFailed",
+			Message: err.Error(),
+		}, nil)
+		return err
+	}
+
+	// make backup job
+	var job *batchv1.Job
+	var reason string
+	if job, reason, err = bc.makeStopLogBackupJob(backup); err != nil {
+		klog.Errorf("backup %s/%s create job %s failed, reason is %s, error %v.", ns, name, stopLogJobName, reason, err)
+		return err
+	}
+
+	// create k8s job
+	if err := bc.deps.JobControl.CreateJob(backup, job); err != nil {
+		errMsg := fmt.Errorf("stop log backup %s/%s job %s failed, err: %v", ns, name, stopLogJobName, err)
+		bc.statusUpdater.Update(backup, &v1alpha1.BackupCondition{
+			Command: v1alpha1.LogStopCommand,
+			Type:    v1alpha1.BackupRetryTheFailed,
+			Status:  corev1.ConditionTrue,
+			Reason:  "StopLogBackupJobFailed",
+			Message: errMsg.Error(),
+		}, nil)
+		return errMsg
+	}
+
+	return bc.statusUpdater.Update(backup, &v1alpha1.BackupCondition{
+		Command: v1alpha1.LogStopCommand,
+		Type:    v1alpha1.BackupScheduled,
+		Status:  corev1.ConditionTrue,
+	}, nil)
+}
+
+func (bc *backupCleaner) CleanData(backup *v1alpha1.Backup) error {
+	if backup.DeletionTimestamp == nil || !v1alpha1.IsCleanCandidate(backup) || v1alpha1.NeedRetainData(backup) {
```

Comment:
For the log backup, we should ensure it is stopped before clean data.

Reply:
Right